pytorch_lightning/trainer/trainer.py

```diff
@@ -995,12 +995,41 @@ def _run_train(self) -> None:
             if distributed_available() and self.world_size > 1:
                 # try syncing remaing processes, kill otherwise
                 self.training_type_plugin.reconciliate_processes(traceback.format_exc())
+            # save a checkpoint for fault tolerant training
+            self.fit_loop._check_checkpoint_callback(True)
```

Comment:
@carmocca that's what I meant the other day when we talked about why we have the `check_checkpoint_callback` in the first place. 

this usage here was the old way Lightning would save checkpoints on interrupt.

Reply:
We shouldnt manually call the `ModelCheckpoint` callback like this as it will be moved to a fully hook-based callback next week.

If training is stopped and we need to save, we should do it by calling the `on_train_end` hook

Reply:
Adding to this:

If we need to manually save a checkpoint, we can always call `trainer.save_checkpoint()` and this doesn't go through the ModelCheckpoint :)

Reply:
I would prefer for the ModelCheckpoint to save the checkpoint as the last_model_path should be the one saved on failure.